src/Npgsql/NpgsqlDataReader.cs

```diff
@@ -1322,7 +1327,7 @@ public override T GetFieldValue<T>(int ordinal)
             }
             catch (NpgsqlSafeReadException e)
             {
-                throw e.InnerException;
+                throw e.OriginalException;
```

Comment:
Any reason not to hide InnerException with a `new` property instead of defining another property?

Reply:
Because it's an original exception, not inner. `NpgsqlSafeReadException` is a container.

Reply:
But you are even passing originalException to innerException in the base class constructor... So the same exception is exposed once via InnerException (nullable) and once via OriginalException (non-nullable)...

Reply:
Removed passing an inner exception to the base class.